packages/adapter-utils/src/utils.ts

```diff
@@ -189,7 +189,12 @@ export const transformValues = async (
         if (strict) {
           log.debug(`Value mis-match for field ${field?.name} - value is not an object`)
         }
-        return (_.isEmpty(newVal) && !allowEmpty) ? undefined : newVal
+        return (_.isEmpty(newVal)
+          // No need to check if for isPlainObject because in
+          // this flow we already checked that it is not
+          && (Array.isArray(newVal)
+            || _.isString(newVal))
+          && !allowEmpty) ? undefined : newVal
```

Comment:
1. are you trying to torture me with this indentation? because it is working
2. the comment is very strange, when I come to read this, why would I even suspect there is a need to check for is object? think about the person that tries to read this code not as a diff with the context of your specific task right now, but just reading the code, sometime in the future, when nobody remembers that at one point we had this one specific bug

```suggestion
        // _.isEmpty returns true for primitive values (boolean, number) but we do not want to omit those
        // We only want to omit empty objects, arrays and strings. we don't need to check for objects here
        // because we cannot get here with an object
        const valueIsEmpty = (Array.isArray(newVal) || _.isString(newVal) && _.isEmpty(newVal)
        return (valueIsEmpty && !allowEmpty) ? undefined : newVal
```